rasterio/dataio.py

```diff
@@ -0,0 +1,61 @@
+"""Classes capable of reading and writing datasets
+
+Instances of these classes are called dataset objects.
```

Comment:
Maybe add a note recommending that instances of these classes be instantiated by `rasterio.open()`